### DISCUSSION TITLE
isActive always false when activating with two injected providers (MetaMask and WalletLink)

### DISCUSSION THREAD

**Discussion by earthtojake** [NoahZinsmeister/web3-react#445] [Q&A]
When there are two injected providers installed in the browser, e.g. MetaMask and WalletLink, and a user accepts one connection and rejects the other (common flow), the rejected connection throws an error and `isActive` state is not set to true for the accepted connector. This also blocks network switch requests which should appear after the connection is accepted. `isActive` state fixes itself on refresh. Can't find a reliable workaround.

This bug should probably be fixed with handling of multiple injected connectors, there's another issue here: https://github.com/NoahZinsmeister/web3-react/issues/300

IMO multiple popups for injected connectors is pretty confusing UX and I don't think it should be the default.

The mustBeMetaMask flag for MM also doesn't fix the issue.

**Comment by earthtojake** [comment]
Update: This is only a problem when you connect to MetaMask with WalletLink installed, but connecting to WalletLink with MetaMask installed works

**Comment by NoahZinsmeister** [comment]
hm, i'm not sure it's worth trying to handle the case when a user has two extensions both injecting to window.ethereum - seems fairly uncommon and kind of an anti-pattern

**Reply by jdbertron** [reply]
Yeah it's worth supporting. I'm developing a contract I don't want to publish to testnet yet, using hardhat, and at the same time I need to issue requests to Ropsten or Mumbai from the DApp. I need two providers.
